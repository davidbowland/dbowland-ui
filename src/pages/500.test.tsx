import ServerErrorMessage from '@components/server-error-message'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { mocked } from 'jest-mock'
import React from 'react'

import InternalServerError from './500'

jest.mock('@aws-amplify/analytics')
jest.mock('@components/server-error-message')

describe('500 error page', () => {
  beforeAll(() => {
    mocked(ServerErrorMessage).mockReturnValue(<></>)
  })

  it('should render ServerErrorMessage', () => {
    const expectedTitle = '500: Internal Server Error'
    render(<InternalServerError />)
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledWith(
      expect.objectContaining({ title: expectedTitle }),
      expect.anything(),
    )
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledTimes(1)
  })
})
