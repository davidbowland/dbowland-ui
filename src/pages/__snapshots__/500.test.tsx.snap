// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`500 error page Ensure snapshot match 1`] = `
<main
  class="sc-dlnjwi gnIxof"
>
  <h1
    class="sc-gtsrHT elXEts"
  >
    500: Internal Server Error
  </h1>
  <p
    class="sc-bdnxRM fSObbe"
  >
    An internal server error has occurred trying to serve your request. If you continue to experience this error, please contact the webmaster.
    <br />
    <br />
    <a
      href="/"
    >
      Go home
    </a>
    .
  </p>
</main>
`;
